databroker/eventsource/shim.py

```diff
@@ -5,18 +5,52 @@
 import pandas as pd
 import logging
 import boltons.cacheutils
+import heapq
 import re
 from ..utils import ALL
+
 # Toolz and CyToolz have identical APIs -- same test suite, docstrings.
 try:
     from cytoolz.dicttoolz import merge
 except ImportError:
     from toolz.dicttoolz import merge
 
-
 logger = logging.getLogger(__name__)
 
 
+def interlace_gens(*gens):
+    """Take generators and interlace their results by timestamp
+
+    Parameters
+    ----------
+    gens : generators
```

Comment:
These docstrings are wrong, the code expects just the dictionaries.

Reply:
No I think this takes in the generators, unless `self.mds.get_events_generator(d)` reports dictionaries.

Reply:
I miss-clicked, this should be on the line below.